ytypes/list_test.go

```diff
@@ -219,7 +219,7 @@ func TestValidateListSimpleKey(t *testing.T) {
 	listSchema := &yang.Entry{
 		Name:     "list-schema",
 		Kind:     yang.DirectoryEntry,
-		ListAttr: yang.NewDefaultListAttr(),
```

Comment:
Why did this change happen? Shouldn't `NewDefaultListAttr()` give us a range of 0->max per [the current implementation](https://github.com/openconfig/goyang/blob/fd7d014c61f43a05d53146582e61503f374787df/pkg/yang/entry.go)?

Reply:
Is the concern here where there is an older implementation of goyang being used? Shouldn't Go modules help us with this?

Reply:
The concern is for any derived package which tries to instantiate a `yang.Entry` where `ListAttr` is needed, then the user may run into the issue where the zero value is not useful, and they have to go and change every place where they instantiate a `yang.Entry` type.

See https://golang.org/doc/effective_go.html#allocation_new
"""
Since the memory returned by new is zeroed, it's helpful to arrange when designing your data structures that the zero value of each type can be used without further initialization. This means a user of the data structure can create one with new and get right to work.
"""